### PR TITLE
Add template configuration options to reference docs in master

### PR DESCRIPTION
**Pull request by dedemorton**
cherry pick of #1183 into master - note that related changes introduced in other PRs have already been merged into master

This PR also changes the default in the docs for template loading from disabled to enabled (see #1135 )

### CODE REVIEW

**Comment by dedemorton** [comment]
@ruflin The second commit addresses your second bullet point in #1135 

"Documentation must be updated to mention the template loading as default and show the shell command as a second option."

File: libbeat/docs/outputconfig.asciidoc [hunk 1]
```diff
@@ -181,6 +181,34 @@ The index root name to write events to. The default is the Beat name.
 For example "{beatname_lc}" generates "[{beatname_lc}-]YYYY.MM.DD" indexes (for example,
 "{beatname_lc}-2015.04.26").
 
+===== template
+
+The http://www.elastic.co/guide/en/elasticsearch/reference/current/indices-templates.html[index
+template] to use for setting mappings in Elasticsearch. By default, template loading is
+enabled. If you disable this option, you must load the template by running the
+load script that is provided with {beatname_uc}.
```
Comment (andrewkroh) [inline libbeat/docs/outputconfig.asciidoc]: AFAIK there isn't a load script, just some commands to execute on the command line. We can link to the instructions for loading the templates.

File: libbeat/docs/shared-template-load.asciidoc [hunk 2]
```diff
@@ -55,14 +58,15 @@ output:
...
-The template is loaded when you start {beatname_uc}. By default, if a template
-already exists in the index, it is not overwritten. To overwrite an existing template,
-set `overwrite: true` in the configuration file.
+By default, if a template already exists in the index, it is not overwritten. To overwrite an existing
+template, set `overwrite: true` in the configuration file.
+
+To disable automatic template loading, comment out the template part under elasticsearch output.
 
 [[load-template-shell]]
 ==== Running a Shell Script to Load the Template
```
Comment (andrewkroh) [inline libbeat/docs/shared-template-load.asciidoc]: Should this say "Shell Command" instead of "Shell Script"?

Comment (dedemorton) [inline libbeat/docs/shared-template-load.asciidoc]: @ruflin Can you look at Andrew's comments and let me know?

Comment (ruflin) [inline libbeat/docs/shared-template-load.asciidoc]: I would also go with `Shell Command`

**Comment by andrewkroh** [comment]
The updates LGTM.

**Comment by dedemorton** [comment]
@ruflin I think this is ready to merge now. (Squashed the review fixes into the second commit.)

**Pull request merged**